### PR TITLE
fix(api-lite): serve account history older than the index from fuel-core

### PR DESCRIPTION
**Pull request by nelitow**
## Summary

Account pages stopped showing transactions older than the api's index window when the account also had a few recent ones. The foundation wallet showed 6 transactions and no page arrows, while fuel-core holds its full history, including August. Account pages now continue from fuel-core right below the oldest indexed transaction, newest first, and the page arrows work in both directions.

## Changes

| Area | Change |
|------|--------|
| Account history | When the index has no older rows for an account, the rest of the page comes from fuel-core, starting below the oldest indexed transaction instead of only when that transaction sat exactly at the index start |
| fuel-core paging direction | Older pages use `last`/`before` and newer pages use `first`/`after`, matching fuel-core on mainnet; the fallback previously returned the account's first-ever transactions |
| Page arrows | `hasPreviousPage` now means older transactions exist and `hasNextPage` means newer ones exist on fuel-core-served pages, as the explorer's pagination expects |
| Disk cache for hot accounts | The blocks behind the last 3 older pages served from fuel-core are kept on disk for the 50 most viewed accounts, so those pages do not download their blocks again |
| Edge cases | A malformed `before` cursor returns the newest page instead of an error, and the total count is never below the rows shown |
| Tests | The fuel-core fake now follows fuel-core's real ordering, plus regression tests for a short index page, a full index page with older history, the last page, and malformed cursors |

### CODE REVIEW

**Comment by nelitow** [comment]
👍

File: packages/api-lite/src/graphql/resolvers/transactions.ts [hunk 1]
```diff
@@ -179,8 +179,13 @@ async function collectDown(
...
 type FcDir =
-  | { kind: 'older'; after?: string }
-  | { kind: 'newer'; before?: string };
+  | { kind: 'older'; before?: string }
+  | { kind: 'newer'; after?: string };
+
+// Height as 8 hex digits, tx index as 4: fixed width, so string order is cursor order.
+export function fuelCoreCursor(height: number, txIndex: number): string {
+  return `${height.toString(16).padStart(8, '0')}${txIndex.toString(16).padStart(4, '0')}`;
+}
```
[the remark this answers is not in the document]
Comment (nelitow) [reply packages/api-lite/src/graphql/resolvers/transactions.ts:188]: Not a bug. fuel-core mainnet returns these cursors without a prefix, e.g. `03cf79700006`, and rejects a prefixed one: `before: "0x03d1f7e10045"` answers `Invalid encoded byte in TxPointer`, while `before: "03d1f7e10045"` returns the older rows. Both checked against https://mainnet.fuel.network/v1/graphql on 2026-09-17. The resolver was also run end to end against that endpoint for the foundation wallet: 4 pages of 10 rows, newest first, no duplicates.

File: packages/api-lite/src/graphql/resolvers/transactions.ts [hunk 2]
```diff
@@ -414,21 +429,22 @@ export const transactionResolvers = {
...
           ctx,
           owner,
           size,
-          { kind: 'older', after: args.before.slice(3) },
+          { kind: 'older', before: args.before.slice(3) },
           [],
           pricing,
         );
-        const total = page.hasNextPage
+        const total = page.moreInDirection
```
[the remark this answers is not in the document]
Comment (nelitow) [reply packages/api-lite/src/graphql/resolvers/transactions.ts:442]: Fixed in 71333865. Both fc: page paths now floor totalCount at the number of rows returned. Test: "never reports a total below the rows on an fc: page".
